### PR TITLE
[Method References 2] LPS-175789 Convert method reference to ServiceTrackerList in class ClusterExecutorImpl

### PR DESCRIPTION
**Pull request by W2jx**

### CODE REVIEW

**Comment by W2jx** [comment]
ci:test:help

**Comment by W2jx** [comment]
ci:test:clustering

File: modules/apps/portal/portal-cluster-multiple/src/test/java/com/liferay/portal/cluster/multiple/internal/ClusterExecutorImplTest.java [hunk 1]
```diff
@@ -66,57 +68,6 @@ public static void setUpClass() {
 		PropsUtil.setProps(new PropsImpl());
 	}
 
-	@Test
-	public void testClusterEventListener() {
```
Comment (dantewang) [inline modules/apps/portal/portal-cluster-multiple/src/test/java/com/liferay/portal/cluster/multiple/internal/ClusterExecutorImplTest.java:70]: @W2jx Please don't just remove the test case; instead, register mocked event listeners to migrate this test case to cover our change.

File: modules/apps/portal/portal-cluster-multiple/src/main/java/com/liferay/portal/cluster/multiple/internal/ClusterExecutorImpl.java [hunk 2]
```diff
@@ -222,19 +203,30 @@ protected void activate(ComponentContext componentContext) {
...
 			new ClusterExecutorPortalInetSocketAddressEventListener(),
 			new HashMapDictionary<String, Object>());
 	}
 
+	@Reference(
+		cardinality = ReferenceCardinality.MULTIPLE,
+		policy = ReferencePolicy.DYNAMIC,
+		policyOption = ReferencePolicyOption.GREEDY
+	)
+	protected void addClusterEventListener(
```
Comment (dantewang) [inline modules/apps/portal/portal-cluster-multiple/src/main/java/com/liferay/portal/cluster/multiple/internal/ClusterExecutorImpl.java]: @W2jx It's weird that you remove the public method with the reference, but add a new protected method with the same name, doing the same thing.

Please do it in this order:

1. Convert direct method references to field references;
2. Convert the `ClusterEventListener` reference to a `ServiceTrackerList`. After this, the public API still exists, but not with any `@Reference` annotation;
3. Remove the public APIs;

File: modules/apps/portal/portal-cluster-multiple/src/main/java/com/liferay/portal/cluster/multiple/internal/ClusterExecutorImpl.java [hunk 3]
```diff
@@ -196,14 +192,18 @@ public boolean isEnabled() {
 	}
 
 	@Activate
-	protected void activate(ComponentContext componentContext) {
+	protected void activate(
+		BundleContext bundleContext, Map<String, String> properties) {
```
Comment (dantewang) [inline modules/apps/portal/portal-cluster-multiple/src/main/java/com/liferay/portal/cluster/multiple/internal/ClusterExecutorImpl.java:196]: @W2jx Why do you need to convert the component context parameter to bundle context + properties map? If it's really needed, please do it in a separate commit.

**Review by dantewang — CHANGES_REQUESTED** [review]
@W2jx Please make changes as I commented, thank you.

**Comment by dantewang** [comment]
ci:stop
